### PR TITLE
[feat] Doughnut charts, Chart Label, Speedometer

### PR DESCRIPTION
**Pull request by raj-kumar-litmus**

### CODE REVIEW

File: src/components/common/ChartLegend.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { LabelProps } from "../../@types/components/commonTypes";
+
+const ChartLegend = (props: LabelProps) => {
+  return (
+    <div className={`flex items-center ${props.containerClassName}`}>
```
Comment (AdithyaShaj) [inline src/components/common/ChartLegend.tsx]: ${props.containerClassName || ""}

File: src/components/common/ChartLegend.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      <span
+        className={`h-[15px] w-[15px] rounded-[50%] inline-block ${props.circleColor}`}
```
Comment (AdithyaShaj) [inline src/components/common/ChartLegend.tsx]: 1. can't we add 15px to tailwind.config?
2. use rounded-full if it is for circle
3. ${props.circleColor || ""}

File: src/components/common/Doughnut.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+);
+
+const DoughnutChart = (props: DoughNutChartProps) => (
+  <div
+    className={`${props.containerClassName} cursor-pointer`}
```
Comment (AdithyaShaj) [inline src/components/common/Doughnut.tsx]: please fix the dynamic class name issue for all the scenarios

File: src/components/common/LinearGauge.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    <span
+      className={`flex items-center justify-center rounded-l-3xl w-[${Math.round(
+        (Number(props.propOne) /
+          (Number(props.propOne) + Number(props.propTwo))) *
+          100,
+      )}%] border-solid border-r-4 border-black-101 h-[${props.height}] ${
+        props.bgColorOne
+      }`}
+    >
+      {Number(props.propOne)}
```
Comment (AdithyaShaj) [inline src/components/common/LinearGauge.tsx:14]: Please check the scenario in which props.propOne is undefined

File: tailwind.config.js [hunk 5]
```diff
@@ -39,7 +40,7 @@ export default {
           105: "#ced4da",
           106: "#9E9E9F",
           107: "#1E1E1F",
-          108: "#424245"
+          108: "#424245",
```
Comment (sarath-sasi-panickaveetil) [inline tailwind.config.js]: seems like #424245 is already defined 'black'

File: src/components/common/LinearGauge.tsx (hunk 4 above)
Comment (raj-kumar-litmus) [reply src/components/common/LinearGauge.tsx:14]: undefined cannot be passed.
https://github.com/raj-kumar-litmus/support-hub/blob/feat/donught-charts/src/%40types/components/commonTypes.tsx#L449-L450

File: tailwind.config.js (hunk 5 above)
Comment (raj-kumar-litmus) [reply tailwind.config.js]: It was thasnim's change. Anyways, I will fix it.

File: src/components/common/Doughnut.tsx (hunk 3 above)
Comment (raj-kumar-litmus) [reply src/components/common/Doughnut.tsx]: For cases involving, optional props I have handled.

File: src/components/common/ChartLegend.tsx (hunk 2 above)
Comment (raj-kumar-litmus) [reply src/components/common/ChartLegend.tsx]: Done.

**Review by AdithyaShaj — APPROVED** [review]

**Pull request merged**